### PR TITLE
[O11y][Apache Spark] Add dimension mapping for application datastream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

This PR adds dimension fields for application data stream to support TSDB enablement.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

- Relates #7786

### CODE REVIEW

**Review by agithomas — APPROVED** [review]
LGTM!

**Comment by agithomas** [comment]
@harnish-elastic , please complete the peer review additionally.

File: packages/apache_spark/docs/README.md [hunk 1]
```diff
@@ -152,10 +148,18 @@ An example event for `application` looks as following:
 | Field | Description | Type |
 |---|---|---|
 | @timestamp | Event timestamp. | date |
+| agent.id | Unique identifier of this agent (if one exists). Example: For Beats this would be beat.id. | keyword |
 | apache_spark.application.cores | Number of cores. | long |
+| apache_spark.application.mbean | The name of the jolokia mbean. | keyword |
 | apache_spark.application.name | Name of the application. | keyword |
```
Comment (agithomas) [inline packages/apache_spark/docs/README.md:154]: Why the application name is not considered to be a dimension field?

**Review by agithomas — APPROVED** [review]
LGTM!

**Review by milan-elastic — APPROVED** [review]
LGTM!

**Pull request merged**